### PR TITLE
Move ElfFile to ElfUtils component

### PR DESCRIPTION
**Pull request by dimitry-**
And out of OrbitCore

### CODE REVIEW

File: ElfUtils/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+add_executable(ElfUtilsTests)
+
+target_sources(ElfUtilsTests PRIVATE
+    ElfFileTest.cpp
+)
+
+target_link_libraries(
+  ElfUtilsTests
+  PRIVATE ElfUtils
+          OrbitCore # TODO: This is needed for GetExecutablePath consider moving Path.h to OrbitBase
```
Comment (dpallotti) [inline ElfUtils/CMakeLists.txt:39]: `Path` would really need to get a serious cleanup before moving it to `OrbitBase` though.

Comment (dpallotti) [inline ElfUtils/CMakeLists.txt:39]:
```suggestion
          OrbitCore # TODO: This is needed for GetExecutablePath - consider moving Path.h to OrbitBase
```

File: OrbitCore/CMakeLists.txt [hunk 2]
```diff
@@ -200,8 +199,8 @@ target_link_libraries(
           llvm_object::llvm_object
           abseil::abseil)
 
-add_custom_command(TARGET OrbitCoreTests POST_BUILD
-  COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_LIST_DIR}/testdata
-  $<TARGET_FILE_DIR:OrbitCoreTests>/testdata)
+# add_custom_command(TARGET OrbitCoreTests POST_BUILD
```
Comment (dpallotti) [inline OrbitCore/CMakeLists.txt:202]: Why is it commented?

File: ElfUtils/CMakeLists.txt (hunk 1 above)
Comment (dimitry-) [reply ElfUtils/CMakeLists.txt:39]: agree I was thinking about moving one function at the time adding tests and rewriting it if necessary.

**Comment by beckerhe** [comment]
What about adding a namespace ElfUtils?

File: OrbitCore/CMakeLists.txt (hunk 2 above)
Comment (dimitry-) [reply OrbitCore/CMakeLists.txt:202]: removed.

**Comment by dimitry-** [comment]
> What about adding a namespace ElfUtils?

Orbit::ElfUtils?

**Comment by beckerhe** [comment]
> > What about adding a namespace ElfUtils?
> 
> Orbit::ElfUtils?

I would prefer not having nested namespaces if we can avoid it. But I'm also fine with Orbit::ElfUtils.

File: ElfUtils/CMakeLists.txt (hunk 1 above)
Comment (dimitry-) [reply ElfUtils/CMakeLists.txt]: done.

**Pull request merged**